src/multisig/multisigTransactionsFactory.ts

```diff
@@ -0,0 +1,509 @@
+import {
+    AddressValue,
+    ArgSerializer,
+    BigUIntValue,
+    ContractFunction,
+    EndpointDefinition,
+    EndpointModifiers,
+    NativeSerializer,
+    OptionType,
+    OptionValue,
+    TokenIdentifierValue,
+    U32Value,
+    U64Type,
+    U64Value,
+    VariadicValue,
+} from "../abi";
+import { TokenComputer, TransactionsFactoryConfig } from "../core";
+import { Address } from "../core/address";
+import { Transaction } from "../core/transaction";
+import { TransactionBuilder } from "../core/transactionBuilder";
+import { SmartContractTransactionsFactory } from "../smartContracts";
+import * as resources from "./resources";
+
+interface IAbi {
+    constructorDefinition: EndpointDefinition;
+    upgradeConstructorDefinition?: EndpointDefinition;
+
+    getEndpoint(name: string | ContractFunction): EndpointDefinition;
```

Comment:
I think type of `name` should only be string. As it is in the current implementation.

Reply:
Maybe we can already deprecate `ContractFunction` in v14. And perhaps we'll remove it in v15.